### PR TITLE
* Display an icon next to the app

### PR DESCRIPTION
**Pull request by louisstow**
Implement issue #390 (https://github.com/mozilla/r2d2b2g/issues/390)

### CODE REVIEW

**Comment by louisstow** [comment]
Just noticed it needs to handle hosted app icons. One tic

**Comment by digitarald** [comment]
Icons can also be data uris

**Comment by mykmelez** [comment]
> Icons can also be data uris

Do you have an example app that uses a data: URL icon? That'd be very helpful for development and testing of this change!

**Comment by mykmelez** [comment]
When I add my Mykzilla testing app to the Dashboard, the icon doesn't appear, and it looks like its URL has the wrong path. The manifest URL is http://mykzilla.org/app/manifest.webapp, and the icon path is /app/img/icon-128.png, but the URL of the icon in the Dashboard is http://mykzilla.org/app/app/img/icon-128.png (with an extra "app/" path part).

File: addon/data/content/js/applist.js [hunk 1]
```diff
@@ -56,7 +56,26 @@ var AppList = (function() {
             app.prettyLastUpdate = timedelta(app.lastUpdate);
         }
         app.prettyType = Simulator.APP_TYPES[app.type];
+        
+        //use a default icon
```
Comment (mykmelez) [inline addon/data/content/js/applist.js:60]: Nit: Mozilla convention (from its [Coding Style Guide](https://developer.mozilla.org/En/Mozilla_Coding_Style_Guide) and usage in this script and elsewhere) is to put a space after comment slashes.

File: addon/data/content/js/applist.js [hunk 2]
```diff
@@ -56,7 +56,26 @@ var AppList = (function() {
...
+        var iconPath = "default.png";
+
+        if (app.icon) {
+            if (app.icon.indexOf("data:image/") === 0) {
+                //set the src to the data uri
+                iconPath = app.icon;
+            }
+            else if (app.type === "hosted") {
```
Comment (mykmelez) [inline addon/data/content/js/applist.js:68]: Nit: Mozilla convention is to cuddle elses, i.e. put them on the same line as the closing brace from the previous _if_ block.

**Comment by mykmelez** [comment]
This looks great, and I only see a couple of very minor nits. I'll merge this with those nits fixed shortly. @louisstow Thanks for the fix!

**Pull request merged**